security/tutorials/calico-policy.md

```diff
@@ -0,0 +1,211 @@
+
+```bash
+wget -q --timeout=5 google.com -O -
+```
+
+It returns the HTML of the google.com home page.
+
+### 2. Lock down all traffic
+
+We will begin by using a default deny [Global Calico Network Policy](https://docs.projectcalico.org/reference/resources/globalnetworkpolicy) (which you can only do using Calico) that will help us adopt best practices in using a [zero trust network model](https://docs.projectcalico.org/security/adopt-zero-trust) to secure our workloads. Note that Global Calico Network Policies don't need a namespace, this will effect all resources that match the selector. Kubernetes Network Policies cannot achieve this by themselves.
+
+```bash
+calicoctl create -f - <<EOF
+apiVersion: projectcalico.org/v3
+kind: GlobalNetworkPolicy
+metadata:
+  name: default-deny
+spec:
+  selector: all()
+  types:
+  - Ingress
+  - Egress
+EOF
+```
+
+#### Verify access - denied all ingress and egress
+
+Because all pods in the namespace are now selected, any ingress traffic which is not explicitly allowed by a policy will be denied.
+
+We can see that this is the case by switching over to our "access" pod in the namespace and attempting to access the nginx service.
+
+```bash
+wget -q --timeout=5 nginx -O -
+```
+
+It will return:
+
+```
+wget: bad address 'google.com'
+```
+
+{: .no-select-button}
+
+Next, try to access google.com.
+
+```bash
+wget -q --timeout=5 google.com -O -
+```
+
+It will return:
+
+```
+wget: bad address 'google.com'
+```
+
+{: .no-select-button}
+
+Now that we have the defalt deny Global Calico Network Policy, all ingress / egress traffic is denied _everywhere_.
+
+### 3. Allow kube-system pods to communicate
+
+Kubernetes uses the kube-system namespace to create pods for the kubernetes cluster to function properly. Let's give this namespace wide permissions to allow these specific pods to function normally. It's important to note here that specific Calico Network Policies are namespaced resources that applies to workloads in that namespace whereas the Calico Global Network Policy is a non-namespaced resource and can be applied to any kind of endpoint (pods, VMs, host interfaces) independent of namespace.
+
+```bash
+calicoctl create -f - <<EOF
+apiVersion: projectcalico.org/v3
+kind: NetworkPolicy
+metadata:
+  name: allow-kube-system
+  namespace: kube-system
+spec:
+  selector: all()
+  types:
+  - Ingress
+  - Egress
+  ingress:
+  - action: Allow
+  egress:
+  - action: Allow
+EOF
+```
+
+### 4. Allow traffic to Nginx from Busybox
+
+Create another Calico Network Policy which allows traffic from the nginx pod from the busybox "access" pod.
+
+```bash
+calicoctl create -f - <<EOF
+apiVersion: projectcalico.org/v3
+kind: NetworkPolicy
+metadata:
+  name: allow-busybox-nginx
+  namespace: advanced-policy-demo
+spec:
+  selector: all()
+  types:
+  - Ingress
+  - Egress
+  ingress:
+  - action: Allow
+    source:
+      selector: run == 'access'
+    destination:
```

Comment:
Rather than having a destination clause in an ingress rule, it is better to make the main selector select the nginx pod rather than all().

Reply:
Should I make a separate policy then which allows egress on the busybox?